### PR TITLE
[Deps] [Redone] Ecore without tests

### PR DESCRIPTION
**Pull request by joaoantoniocardoso**
After #159 Ecore will be building fine **without tests**.

However before this PR is ready for review we need to:
  - [x] gettimeofday into native-windows (fixed by #159)
  - [x] properly fix winsock2 (fixed by #197)

To be clear why **without tests**, to enable test to be built we'll need to build:
  - ecore_file  
     - ecore_con  
  - ecore_input  
  - ecore_evas  
    - ecore_input_evas  
       - evas  
       - ecore_win32
  - ecore_imf  
  - _and possibly others..._

### CODE REVIEW

**Comment by Coquinho** [comment]
> To be clear why **without tests**, to enable test to be built we'll need to build:
> 
> * ecore_file
>   
>   * ecore_con
> * ecore_input
> * ecore_evas
>   
>   * ecore_input_evas
>     
>     * evas
>     * ecore_win32
> * ecore_imf
> * _and possibly others..._

evas-fixes has ecore tests running, so it won't be a long wait.

File: src/lib/ecore/ecore_time.c [hunk 1]
```diff
@@ -55,14 +55,14 @@ ecore_time_get(void)
...
    struct timeval timev;
 
    gettimeofday(&timev, NULL);
    return (double)timev.tv_sec + (((double)timev.tv_usec) / 1000000);
-#else
-# error "Your platform isn't supported yet"
-#endif
+//#else
+//# error "Your platform isn't supported yet"
+//#endif
```
Comment (Coquinho) [inline src/lib/ecore/ecore_time.c:65]: Instead of commenting all this, you could make this be a warning instead of an error.

File: src/lib/evil/evil_strings.h [hunk 2]
```diff
@@ -2,14 +2,9 @@
...
+#include <evil_windows.h>
 
-#ifdef _WIN32
-# ifndef WIN32_LEAN_AND_MEAN
-#  define WIN32_LEAN_AND_MEAN
-# endif
-# include <windows.h>
-# undef WIN32_LEAN_AND_MEAN
-#else
+#ifndef _WIN32
```
Comment (Coquinho) [inline src/lib/evil/evil_strings.h:7]: This probably should be `_MSC_VER` as evil only runs on windows.

**Comment by joaoantoniocardoso** [comment]
As soon as #159 merges the ecore will be building.

File: src/lib/ecore/ecore_time.c [hunk 3]
```diff
@@ -61,7 +61,7 @@ ecore_time_unix_get(void)
    gettimeofday(&timev, NULL);
    return (double)timev.tv_sec + (((double)timev.tv_usec) / 1000000);
 #else
-# error "Your platform isn't supported yet"
+# warning "Your platform isn't supported yet"
```
Comment (Coquinho) [inline src/lib/ecore/ecore_time.c:64]: why is this change needed?

Comment (joaoantoniocardoso) [reply src/lib/ecore/ecore_time.c:64]: I maybe got it wrong but wasn't your [suggestion](https://github.com/expertisesolutions/efl/pull/294#discussion_r453602013)?

Comment (Coquinho) [reply src/lib/ecore/ecore_time.c:64]: for while `gettimeofday` wasn't being implemented, yes. Now that we have it and we define `HAVE_GETTIMEOFDAY` there is no reason to maintain this change.
